### PR TITLE
assert.h: Replace NULL with 0 to avoid including stddef.h

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

Fix error: 'NULL' undeclared if user include assert.h only

## Impact

Minor

## Testing

CI

### CODE REVIEW

File: include/assert.h [hunk 1]
```diff
@@ -66,20 +66,20 @@
...
-  do                                        \
-    {                                       \
-      if (predict_false(!(f)))              \
-        __assert(__FILE__, __LINE__, NULL); \
-    }                                       \
+#  define ASSERT(f)                       \
+  do                                      \
+    {                                     \
+      if (predict_false(!(f)))            \
+        __assert(__FILE__, __LINE__, 0);  \
```
Comment (pkarashchenko) [inline include/assert.h:73]: Will this not lead to an error that integer is passed where the pointer is expected?

Comment (xiaoxiang781216) [reply include/assert.h:73]: it's valid to pass 0 to pointer argument.

Comment (pkarashchenko) [reply include/assert.h:73]: Ok. I just checked. The parameter values are not checked. I was thinking that `-Wint-conversion` will catch it, but seems it handles only assignment cases (and passing a parameter is a kind of indirect assignment, so strange why this is not caught).

**Review by pkarashchenko — APPROVED** [review]

**Review by patacongo — APPROVED** [review]
LGTM

**Pull request merged**